### DISCUSSION TITLE
Option to Disable "Commit Suggestion" Feature in Pull Requests

### DISCUSSION THREAD

**Discussion by gfringeli** [community/community#137816] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

It would be helpful to have an option in repository settings to disable the "Commit Suggestion" feature during code reviews. While this feature is useful for many workflows, there are cases where repository maintainers may want to enforce stricter control over how commits are made to the repository.

## Problem Statement
In some repositories, especially those with rigorous code quality standards or a specific commit history policy, allowing direct commits via the "Commit Suggestion" button can lead to:

- Commit Noise: Frequent small commits that clutter the history and make it harder to understand the project's development.
- Unwanted Direct Changes: Bypassing local testing, CI checks, or other automated validation processes that are usually enforced when contributors make changes locally.
- Reduced Control Over Code Quality: Maintainers lose some control over the exact changes being merged if suggestions are committed directly without adequate review.

## Proposed Solution:
Introduce a repository setting or branch protection rule that allows maintainers to disable the "Commit Suggestion" feature for specific branches or the entire repository. This setting would allow all other code review features to remain active while preventing direct commits via the suggestion button.
Benefits:

- Improved Control: Maintainers can enforce desired workflows, such as requiring local testing or more comprehensive reviews before commits.
- Cleaner Commit History: Helps maintain a clean and meaningful commit history, which is especially important in larger projects or those following a specific commit convention.
- Consistency with Development Practices: Ensures that all commits to the repository go through the same vetting process, enhancing the consistency and quality of the codebase.

## Conclusion
Adding an option to disable the "Commit Suggestion" feature would give repository maintainers more control over the commit process, aligning with diverse workflows and standards. This flexibility would improve GitHub's usability for teams with strict code review and quality requirements.

**Comment by masterwww1** [comment]
I would like to propose a feature enhancement for GitHub’s code review process: the ability to disable the "Commit Suggestion" feature in repository settings.

Problem Statement: In some development workflows, especially those with strict code quality requirements or specific commit policies, the "Commit Suggestion" feature can lead to several issues:

Commit Noise: Small and frequent commits that can clutter the commit history.
Unwanted Direct Changes: Direct commits can bypass local testing and automated CI/CD checks.
Reduced Quality Control: Suggestions committed directly might not undergo the same rigorous review process.
Proposed Solution: Introduce a repository setting or branch protection rule that allows maintainers to disable the "Commit Suggestion" feature for certain branches or the entire repository. This would ensure that all commits are subject to the same review and testing processes.

Benefits:

Enhanced Workflow Control: Maintainers can enforce thorough review processes and local testing requirements before changes are committed.
Cleaner Commit History: Helps maintain a clear and organized commit history.
Consistency: Ensures that all changes adhere to the same quality standards and review practices.
Adding an option to disable the "Commit Suggestion" feature would greatly benefit teams with stringent code quality standards by providing more control over the commit process. This would align better with diverse development workflows and improve overall code management.
